document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/AbridgedAccountsDataHandler.java

```diff
@@ -142,21 +147,21 @@ private String createPathString(AccountType accountType) {
         return String.format("/%s/%s", accountType.getAssetId(), accountType.getUniqueFileName());
     }
 
-    private String createDocumentInfoResponseData(Transaction transaction, AbridgedAccountsApi accountData,
-                                                  AccountType accountType) throws ServiceException {
-
-        String accountTypeName = accountType.getResourceKey();
+    private String createDocumentInfoResponseData(Transaction transaction, AbridgedAccountsApi accountData)
+            throws ServiceException, IOException {
 
-        JSONObject accountJSON = new JSONObject(accountData);
-        JSONObject account = new JSONObject();
-
-        account.put(accountTypeName, accountJSON);
-        account.put("company_number", transaction.getCompanyNumber());
+        AbridgedAccountsApiData abridgedAccountsApiData = new AbridgedAccountsApiData();
+        abridgedAccountsApiData.setAbridgedAccountsApi(accountData);
 
         CompanyProfileApi companyProfile = companyService.getCompanyProfile(transaction.getCompanyNumber());
-        account.put("company_name", companyProfile.getCompanyName());
+        abridgedAccountsApiData.setCompanyName(companyProfile.getCompanyName());
+        abridgedAccountsApiData.setCompanyNumber(companyProfile.getCompanyNumber());
+
+        ObjectMapper mapper = new ObjectMapper();
+        String accountsJSON = mapper.writeValueAsString(abridgedAccountsApiData);
+        JsonNode accounts = mapper.readTree(accountsJSON);
```

Comment:
We could move this block  to different method following the Single Responsibility Principle.  
```
ObjectMapper mapper = new ObjectMapper();
String accountsJSON = mapper.writeValueAsString(abridgedAccountsApiData);
JsonNode accounts = mapper.readTree(accountsJSON);
```

Reply:
done